lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -147,7 +149,7 @@ private synchronized void onRMDone(final RuntimeStatusEvent runtimeStatusEvent)
     this.driverStatusManager.onComplete();
   }
 
-  private synchronized void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
+  private void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
```

Comment:
I have run tests with 500 nodes for 20 times and the results consistently show that the fix has resolved the issue.